### PR TITLE
Provides API endpoint to retreive a recovery phrase

### PR DESCRIPTION
**Pull request by akeem**
# Description

Adding test coverage around user Operations

While writing this, a way to approach handling request details for users has started to take shape and will be tackled next :-)

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: locksmith/__tests__/controllers/userController.test.ts [hunk 1]
```diff
@@ -101,7 +103,29 @@ describe('User Controller', () => {
...
+        expect(response.statusCode).toBe(400)
+      })
+    })
+  })
+
+  describe("retrieving a user's recovery phrase", () => {
+    describe('when the user exists', () => {
+      it("returns the user's recovery phrase", async () => {
+        let response = await request(app).get(
+          '/users/user@example.com/recovery_phrase'
```
Comment (julien51) [inline locksmith/__tests__/controllers/userController.test.ts]: since we use `privatekey`, can we use `recoveryphrase`?

Comment (julien51) [reply locksmith/__tests__/controllers/userController.test.ts]: I know it is probably sub-optimal, but I'd rather err on the side of consistency.

File: locksmith/__tests__/controllers/userController.test.ts [hunk 2]
```diff
@@ -101,7 +103,29 @@ describe('User Controller', () => {
...
+        )
+        expect(response.body).toEqual({ recoveryPhrase: 'a recovery phrase' })
+      })
+    })
+
+    describe('when the user does not exist', () => {
```
Comment (julien51) [inline locksmith/__tests__/controllers/userController.test.ts]: I think similarly to how we server a "fake" encrypted private key we should serve a fake recovery key here if the user does not exist.

File: locksmith/src/controllers/userController.ts [hunk 3]
```diff
@@ -27,14 +27,31 @@ namespace UserController {
...
+      return res.json({ recoveryPhrase: result })
+    } else {
+      return res.sendStatus(400)
+    }
+  }
 }
 
-export = UserController
\ No newline at end of file
+export = UserController
```
Comment (julien51) [inline locksmith/src/controllers/userController.ts]: I think we have linting/prettifier errors in the locksmith project that we need to solve...

File: locksmith/__tests__/controllers/userController.test.ts (hunk 2 above)
Comment (akeem) [reply locksmith/__tests__/controllers/userController.test.ts]: Agreed, I've noted this as a next step in the description of this pr!

Comment (julien51) [reply locksmith/__tests__/controllers/userController.test.ts]: oups! sorry about missing it...

File: locksmith/src/controllers/userController.ts (hunk 3 above)
Comment (akeem) [reply locksmith/src/controllers/userController.ts]: will chat with Chris about this on monday!

**Pull request merged**